sdk/quantum/Azure.Quantum.Jobs/tests/QuantumJobClientLiveTests.cs

```diff
@@ -22,64 +24,147 @@ public QuantumJobClientLiveTests(bool isAsync) : base(isAsync)
-                    Assert.AreEqual("microsoft.qio.v2", job.InputDataFormat);
-                    Assert.AreEqual("Sanitized", job.InputDataUri);
-                    Assert.AreEqual("first-demo", job.Name);
-                    Assert.AreEqual("microsoft.qio-results.v2", job.OutputDataFormat);
-                    Assert.AreEqual("Sanitized", job.OutputDataUri);
-                    Assert.AreEqual("microsoft", job.ProviderId);
-                    Assert.AreEqual(JobStatus.Succeeded, job.Status);
-                    Assert.AreEqual("microsoft.paralleltempering-parameterfree.cpu", job.Target);
-                }
-                else if (index == 1)
+                    BlobName = blobName,
+                })).Value.SasUri;
+
+            // Upload input data to blob (if not in Playback mode)
+            if (Mode != RecordedTestMode.Playback)
+            {
+                var blobClient = new BlobClient(new Uri(inputDataUri));
+                var problemFilename = Path.Combine(Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location), "problem.json");
+                await blobClient.UploadAsync(problemFilename, overwrite: true);
+            }
+
+            // Submit job
+            var jobId = $"job-{TestEnvironment.GetRandomId("JobId")}";
+            var jobName = $"jobName-{TestEnvironment.GetRandomId("JobName")}";
+            var inputDataFormat = "microsoft.qio.v2";
+            var outputDataFormat = "microsoft.qio-results.v2";
+            var providerId = "microsoft";
+            var target = "microsoft.paralleltempering-parameterfree.cpu";
+            var createJobDetails = new JobDetails(containerUri, inputDataFormat, providerId, target)
+            {
+                Id = jobId,
+                InputDataUri = inputDataUri,
+                Name = jobName,
+                OutputDataFormat = outputDataFormat
+            };
+            var jobDetails = (await client.CreateJobAsync(jobId, createJobDetails)).Value;
+
+            // Check if job was created correctly
+            Assert.AreEqual(inputDataFormat, jobDetails.InputDataFormat);
+            Assert.AreEqual(outputDataFormat, jobDetails.OutputDataFormat);
+            Assert.AreEqual(providerId, jobDetails.ProviderId);
+            Assert.AreEqual(target, jobDetails.Target);
+            Assert.IsNotEmpty(jobDetails.Id);
+            Assert.IsNotEmpty(jobDetails.Name);
+            Assert.IsNotEmpty(jobDetails.InputDataUri);
+            if (Mode == RecordedTestMode.Playback)
+            {
+                Assert.IsTrue(jobDetails.Id.StartsWith("job-"));
+                Assert.IsTrue(jobDetails.Name.StartsWith("jobName-"));
+            }
+            else
+            {
+                Assert.AreEqual(jobId, jobDetails.Id);
+                Assert.AreEqual(jobName, jobDetails.Name);
+                Assert.AreEqual(inputDataUri, jobDetails.InputDataUri);
+            }
```

Comment:
Why is this different during playback?  Use `this.Recording.Random` if you want random values while recording but the same values when playing back.